python/lsst/daf/butler/core/datastoreRecordData.py

```diff
@@ -41,24 +41,24 @@
 if TYPE_CHECKING:
     from ..registry import Registry
 
-_Record = Dict[str, Any]
+_Record = dict[str, Any]
 
 
 class SerializedDatastoreRecordData(BaseModel):
     """Representation of a `DatastoreRecordData` suitable for serialization."""
 
-    dataset_ids: List[uuid.UUID]
+    dataset_ids: list[uuid.UUID]
     """List of dataset IDs"""
 
-    records: Dict[str, Dict[str, List[_Record]]]
+    records: Mapping[str, Mapping[str, list[_Record]]]
```

Comment:
Isn't it actually a dict though? What's the gain from using Mapping here?

Reply:
Indeed, the implementation uses `dict` but the clients do not need/want to modify it, so for everyone else an immutable mapping should be OK. I just miss `const` from C++ 🙂

Reply:
Aha. I understand. We are using the annotation to make sure that mypy complains if someone tries to write to the dict.